### ISSUE TITLE
Inertia form helper cast attributes to string

### ISSUE THREAD

**Issue by lucraraujo** [inertiajs/inertia#455]
### Versions:

- `@inertiajs/inertia` version: 0.8.2
- `@inertiajs/inertia-vue` version: 0.5.4

### Describe the problem:

I'm using Buefy (Bulma for Vue.js)
I have a edit page that receives data from Laravel backend, and in this form, I have date fields and the buefy datepicker component accepts Date, or Array.
In the data page attribute i'm initializing all the data that comes from Laravel, including creating a new date object for that date field.
The problem is that, when the page is rendered, the date attribute is casted to string causing the datepicker to throw an error.
The expected result is that the form helper does not cast to string the attributes.

### Steps to reproduce:

```vue
<template>
    <section class="section">
        <div class="container">
            <p class="title">Edit page</p>
            <form>
                <b-field label="This date field don't work">
                    <b-datepicker
                            v-model="form.thisDateDontWork"
                            editable
                            required
                    />
                </b-field>
                <b-field label="This date works">
                    <b-datepicker
                            v-model="thisDateWorks"
                            editable
                            required
                    />
                </b-field>
            </form>
        </div>
    </section>
</template>

<script>
  export default {
    data() {
      const backendData = {
        date: "2021-03-05T03:00:00.000000Z"
      }
      
      return {
        form: this.$inertia.form({
          thisDateDontWork: new Date(backendData.date)
        }),
        thisDateWorks: new Date(backendData.date)
      }
    }
  }
</script>
```

**Comment by ajnsn** [comment]
@lucraraujo This is something I also already noticed recently. I submitted a PR for it.

**Comment by reinink** [comment]
Thanks for reporting! This has been fixed in #575. 🙌
